### DISCUSSION TITLE
Should we start using squash merging?

### DISCUSSION THREAD

**Discussion by amacado** [devicons/devicon#470] [Q&A] [answered]
What do you think about [squash merging](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#squash-and-merge-your-pull-request-commits)? I think it would result in a cleaner commit history where it's easier to find out which icon was added when. I did squash merging for the last few icon pull requests and as you can see in the history it's very easy to check which icon was added when. 

All those "mini-fixes" with undescriptive commit messages only make things harder to read (see two samples I marked here):

![image](https://user-images.githubusercontent.com/8781699/103951615-52456080-513f-11eb-8fef-d060960d9d78.png)

This is what a squashed merge result look like:
![image](https://user-images.githubusercontent.com/8781699/103952072-0810af00-5140-11eb-9404-bc69c64824b4.png)

I would not disable other merge methods, since there could be reasons to use them for certain pull requests, but in general I would suggest to squash merge (especially icon-related pull requests).

![image](https://user-images.githubusercontent.com/8781699/103951993-ed3e3a80-513f-11eb-94d0-ee09155cf71f.png)

**Comment by Thomas-Boi** [comment]
That looks like a good idea. Will this behaviour be done automatically by GitHub or do we have to manually do it?

**Reply by amacado** [reply]
You have to select it while merging a pull request. GitHub will keep your latest selection as default value for the next time:

![image](https://user-images.githubusercontent.com/8781699/104089652-da775300-5270-11eb-8173-bfe89f8b9188.png)

As I mentioned before, we CAN enforce GitHub to only provide the option to squash merge, but I would not suggest to enforce it.

**Reply by Thomas-Boi** [reply]
Ah ok, that's a pretty flexible option then. I'll do that for future PRs then.

**Comment by amacado** [comment, marked as answer]
TL;TR;
We are using squash merging on future pull requests. We will not enforce this setting but recommend it.
